File: app/src/main/java/com/example/geosolve/MainActivity.java
package com.example.geosolve;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;

import androidx.appcompat.app.AppCompatActivity;

import com.example.geosolve.ActivityFragment.SolveActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ImageButton solve_button = findViewById(R.id.solve_button);
        solve_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                 RuleBase.solve(DrawFragment.drawCanvas);
                Intent intent = new Intent(MainActivity.this, SolveActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/example/geosolve/MainActivity.kt
package com.example.geosolve

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.Navigation

class MainActivity : AppCompatActivity() {

    companion object {
        var presenter: Presenter = Presenter()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        presenter.navController = Navigation.findNavController(this, R.id.nav_host_fragment)

    }
}
